### PR TITLE
SBT Update

### PR DESCRIPTION
**Pull request by ArquintL**
Updates SBT to version 1.4.4.

This PR is based on #146.

@Felalolf: You might want to look at commit 5cd4a504d9067571ba7dc92122530f5623449931 in which I had to remove `breakOut`. I hope I did it right (and without performance penalties) but please double check by searching for `breakOut`.

@jcp19: You might want to look at commit 8a2c1bc83982920204d7d3382c7169b86690a863, specifically `assertionExprInBounds(expr, typ)` in `OverflowChecksTransform.scala`. I noticed that the parameter `typ` was not used in your implementation so far and I thought this was a small mistake. I fixed it by changing the set of expressions to a set of tuples with expressions and their associated type. Hence, I can use the input `typ` for the outer most expression. I hope this is what you originally intended.

### CODE REVIEW

**Review by Felalolf — APPROVED** [review]
Great work!

**Comment by Felalolf** [comment]
There are still some `right.toOptions` in the set encoding and in the viper writer files. Also, wow these are a lot of compiler warnings. Especially the "never used" warnings are quite annoying. Can we maybe deactivate these in the SBT file? You do not have to do this in this PR, but someone could check this.

**Review by jcp19 — APPROVED** [review]

File: src/main/scala/viper/gobra/frontend/info/base/SymbolTable.scala [hunk 1]
```diff
@@ -7,11 +7,32 @@
...
+    * A entity represented by names for whom we have seen more than one
+    * declaration so we are unsure what is being represented.
+    */
+  case class MultipleEntity() extends ErrorEntity
+
+  /**
+    * An unknown entity, for example one that is represened by names whose
+    * declarations are missing.
+    */
+  case class UnknownEntity() extends ErrorEntity
```
[the remark this answers is not in the document]
Comment (ArquintL) [reply src/main/scala/viper/gobra/frontend/info/base/SymbolTable.scala:35]: Ah that makes sense. The other thing I noticed was that Environment now takes a type parameter which has to be a subtype of `Entity`. I'm simply using `Entity` there but the Kiama examples seem to use an abstract subclass specialized to the language / use case

File: src/main/scala/viper/gobra/frontend/info/implementation/resolution/AmbiguityResolution.scala [hunk 2]
```diff
@@ -42,7 +42,7 @@ trait AmbiguityResolution { this: TypeInfoImpl =>
   }
 
   def asExpr(n: PExpressionOrType): Option[PExpression] = exprOrType(n).left.toOption
-  def asType(n: PExpressionOrType): Option[PType] = exprOrType(n).right.toOption
+  def asType(n: PExpressionOrType): Option[PType] = exprOrType(n).toOption
```
[the remark this answers is not in the document]
Comment (ArquintL) [reply src/main/scala/viper/gobra/frontend/info/implementation/resolution/AmbiguityResolution.scala:45]: Correct, `Either` does no longer consider left and right as equally important but "prefers" right

**Comment by ArquintL** [comment]
Thanks a lot for the super quick review!

**Pull request merged**